### ISSUE TITLE
All KomodoObject Subclasses Should Overwrite getParent(UnitOfWork) Method To Return Specific Type

### ISSUE THREAD

**Issue by elvisisking** [Teiid-Designer/komodo#314]
If a `KomodoObject` subclass has a known parent type, the `getParent` method should be overridden to return that specific type.

One example is the `Mask` class. It should provide this method:
```
@Override
Permission getParent( final UnitOfWork transaction ) throws Exception;
```

**Comment by mmakovy** [comment]
Is the example correct ? 
As I see it in the source code parent of **Mask** is **RelationalObject** and parent of **MaskImpl** is **RelationalChildRestrictedObject**. Am I right or am I missing something?

**Comment by elvisisking** [comment]
Take a look at [RelationalModelFactory](https://github.com/Teiid-Designer/komodo/blob/master/plugins/org.komodo.relational/src/org/komodo/relational/RelationalModelFactory.java#L410) and how a `Mask` is created. Notice that a `Permission` is passed in as the parent.

So you can add this code to the `Mask` interface: 
```
    /**
     * {@inheritDoc}
     *
     * @see org.komodo.spi.repository.KNode#getParent(org.komodo.spi.repository.Repository.UnitOfWork)
     */
    @Override
    Permission getParent( final UnitOfWork transaction ) throws KException;
```
And then this code to `MaskImpl`:
```
    /**
     * {@inheritDoc}
     *
     * @see org.komodo.relational.internal.RelationalObjectImpl#getParent(org.komodo.spi.repository.Repository.UnitOfWork)
     */
    @Override
    public Permission getParent( final UnitOfWork transaction ) throws KException {
        ArgCheck.isNotNull( transaction, "transaction" ); //$NON-NLS-1$
        ArgCheck.isTrue( ( transaction.getState() == State.NOT_STARTED ), "transaction state must be NOT_STARTED" ); //$NON-NLS-1$

        final KomodoObject grouping = super.getParent( transaction );
        final Permission result = Permission.RESOLVER.resolve( transaction, grouping.getParent( transaction ) );
        return result;
    }
```
Most types don't have a grouping node so the code should be simpler than this. Let me know if I can help.

**Comment by elvisisking** [comment]
Fixed by #333
